lib/dotiw/time_hash.rb

```diff
@@ -114,10 +114,17 @@ def build_years_months_weeks_days
         days_in_month = Time.days_in_month(last_month.month, last_month.year)
         weeks += days_in_month / 7
         days += days_in_month % 7
+
         if days >= 7
           days -= 7
           weeks += 1
         end
+
+        if weeks < 0
+          months -= 1
+          weeks += 5
```

Comment:
Here `weeks` can only equal to `-1` (as I can see). 
So, we should decrement `months` and `-1` should became `4`.